packages/compass-e2e-tests/tests/atlas-login.test.ts

```diff
@@ -0,0 +1,425 @@
+      await cancelButton.click();
+
+      const aiInput = browser.$(Selectors.QueryBarAITextInput);
+      expect(await aiInput.isExisting()).to.eq(false);
+      expect(await generateQueryButton.isDisplayed()).to.eq(true);
+    });
+
+    it('should hide AI input if declined TOS after sign in', async function () {
+      const generateQueryButton = browser.$('button*=Generate query');
+      await generateQueryButton.waitForClickable();
+      await generateQueryButton.click();
+
+      const logInButton = browser.$(Selectors.LogInWithAtlasModalButton);
+      await logInButton.waitForClickable();
+      await logInButton.click();
+
+      const agreeAndContinueButton = browser.$(
+        Selectors.AgreeAndContinueButton
+      );
+      await agreeAndContinueButton.waitForClickable();
+      await agreeAndContinueButton.click();
+
+      await browser.openSettingsModal('Feature Preview');
+
+      const acceptTOSToggle = browser.$(Selectors.AcceptTOSToggle);
+      await acceptTOSToggle.waitForClickable();
+      await acceptTOSToggle.click();
+
+      expect(await acceptTOSToggle.getAttribute('aria-checked')).to.eq('false');
+
+      await browser.closeSettingsModal();
+
+      const aiInput = browser.$(Selectors.QueryBarAITextInput);
+      expect(await aiInput.isExisting()).to.eq(false);
+      expect(await generateQueryButton.isDisplayed()).to.eq(true);
+    });
+  });
+
+  describe('in Aggregation Builder view', function () {
+    beforeEach(async function () {
+      await createNumbersCollection();
+      await browser.connectWithConnectionString();
+      await browser.navigateToCollectionTab('test', 'numbers', 'Aggregations');
+    });
+
+    it('should allow to sign in and accept TOS when clicking AI CTA', async function () {
+      const generateQueryButton = browser.$('button*=Generate aggregation');
+      await generateQueryButton.waitForClickable();
+      await generateQueryButton.click();
+
+      const logInButton = browser.$(Selectors.LogInWithAtlasModalButton);
+      await logInButton.waitForClickable();
+      await logInButton.click();
+
+      const agreeAndContinueButton = browser.$(
+        Selectors.AgreeAndContinueButton
+      );
+      await agreeAndContinueButton.waitForClickable();
+      await agreeAndContinueButton.click();
+
+      // If the flow failed, we will not see the input
+      const aiInput = browser.$(Selectors.QueryBarAITextInput);
+      await aiInput.waitForDisplayed();
+    });
+
+    it('should not show AI input if sign in flow was not finished', async function () {
+      getTokenPayload = () => {
+        return new Promise(() => {});
+      };
+
+      const generateQueryButton = browser.$('button*=Generate aggregation');
+      await generateQueryButton.waitForClickable();
+      await generateQueryButton.click();
+
+      const logInButton = browser.$(Selectors.LogInWithAtlasModalButton);
+      await logInButton.waitForClickable();
+      await logInButton.click();
```

Comment:
Just curious, this is a common pattern in the file here, but how does this differ from `await browser.clickVisible(Selectors.LogInWithAtlasModalButton)` etc. in practice?

Reply:
We do more checks in `clickVisible` (it's our custom method compared to this `waitForClickable` that is part of wdio api), but it doesn't accept webdriver elements as argument, only selector strings, and I need to keep references to the element in some test cases, so for consistency I used it in most places here. Let me see how hard it would be to add support for passing elements in addition to strings in the method, probably better to stick to the one we use everywhere else

Reply:
Updated in https://github.com/mongodb-js/compass/pull/4900/commits/c89a4e09004f9437441e74f7775c3619ef17076a